sdk/eventgrid/Microsoft.Azure.Messaging.EventGrid.CloudNativeCloudEvents/src/EventGridPublisherClientExtensions.cs

```diff
@@ -90,19 +92,19 @@ public static class EventGridPublisherClientExtensions
             writer.WriteStartArray();
             foreach (var cloudEvent in cloudEvents)
             {
-                var attributes = cloudEvent.GetAttributes();
+                List<string> populatedAttributes = cloudEvent.GetPopulatedAttributes().Select(pair => pair.Key.Name).ToList();
                 if (activityId != null &&
-                        !attributes.ContainsKey(TraceParentHeaderName) &&
-                        !attributes.ContainsKey(TraceStateHeaderName))
+                    !populatedAttributes.Contains(TraceParentHeaderName) &&
+                    !populatedAttributes.Contains(TraceStateHeaderName))
```

Comment:
```suggestion
                    !(cloudEvent.GetPopulatedAttributes().Any(attr => attr.Name == TraceParentHaderName || attr.Name == TraceStateHeaderName))
```

Not sure if you're interested only in an exact match or if doing a `string.Equals` for an ordinal case-insensitive comparison would be better...

Reply:
The attributes have to be lowercase so it shouldn't matter.